### PR TITLE
Define GUID for boot manage policy to connect storage devices.

### PR DESCRIPTION
**Pull request by cfernald**
## Description

Some devices may require connecting storage media while booting to network or require enumerating storage protocols that were not initially enumerated during DXE. This change adds a GUID to allow implementation of boot manager's ConnectDeviceClass to connect storage media.

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

N/A

## Integration Instructions

N/A

### CODE REVIEW

File: MdePkg/Include/Protocol/BootManagerPolicy.h [hunk 1]
```diff
@@ -31,6 +31,13 @@
     0x113B2126, 0xFC8A, 0x11E3, { 0xBD, 0x6C, 0xB8, 0xE8, 0x56, 0x2C, 0xBA, 0xFA } \
   }
 
+// MU_CHANGE START: Introduce disk connect guid
+#define EFI_BOOT_MANAGER_POLICY_STORAGE_GUID \
```
Comment (kuqin12) [inline MdePkg/Include/Protocol/BootManagerPolicy.h:35]: Probably also need `gEfiBootManagerPolicyConnectStorageGuid`

Comment (kuqin12) [reply MdePkg/Include/Protocol/BootManagerPolicy.h:35]: And also similar change in the dec file?

**Review by kuqin12 — APPROVED** [review]

File: MdePkg/Include/Protocol/BootManagerPolicy.h (hunk 1 above)
Comment (cfernald) [reply MdePkg/Include/Protocol/BootManagerPolicy.h:35]: good point

**Review by apop5 — APPROVED** [review]

File: MdePkg/Include/Protocol/BootManagerPolicy.h (hunk 1 above)
Comment (makubacki) [inline MdePkg/Include/Protocol/BootManagerPolicy.h:35]: Is this planned to go to the UEFI spec?

Comment (cfernald) [reply MdePkg/Include/Protocol/BootManagerPolicy.h:35]: Yes, we are planning to take this to the UEFI spec and ideally push an open source implementation to EDK2's boot manager policy module

**Review by makubacki — APPROVED** [review]

File: MdePkg/Include/Protocol/BootManagerPolicy.h (hunk 1 above)
Comment (makubacki) [reply MdePkg/Include/Protocol/BootManagerPolicy.h:35]: Thanks. In that case, `EFI_` makes sense as a code first change and we'd get parity again with the upstream module.

**Pull request merged**